src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -82,7 +82,9 @@ public function warmUp($cacheDir)
             return;
         }
 
-        foreach ($this->resourceLocales as $locale) {
+        $locales = array_merge(array($this->getLocale()), $this->getFallbackLocales(), $this->resourceLocales);
+        foreach (array_unique($locales) as $locale) {
+            $this->catalogues = array();
```

Comment:
I found a circular reference which is fixed in #15527 and for resseting catalogues seems only required when generation catalogue for fallback locales.